common/djangoapps/xblock_django/plugin_translations.py

```diff
@@ -0,0 +1,209 @@
+"""
+Plugin translations pulling and compilation logic.
+
+TODO: Move to new openedx.core module to support non-XBlock translations???
+"""
+
+import os
+import shutil
+import subprocess
+import logging
+
+from django.conf import settings
+from django.utils.encoding import force_str
+from django.views.i18n import JavaScriptCatalog
+from django.contrib.staticfiles.storage import staticfiles_storage
+from django.utils.translation import override
+
+from .api import get_xblocks
+from .settings import (
+    XBLOCK_TRANSLATIONS_ROOT,
+    XBLOCK_JS_TRANSLATIONS_RELATIVE_DIR,
+)
+from .constants import ATTR_I18N_JS_NAMESPACE
+
+
+log = logging.getLogger(__name__)
+
+
+class AtlasJavaScriptCatalog(JavaScriptCatalog):
+    """
+    View to return the selected language catalog as a JavaScript library.
+
+    This is a copy to the JavaScriptCatalog class while using a non-standard locale path.
+    """
+    locale_paths = None
+
+    def get_paths(self, packages):
+        return self.locale_paths
+
+    @classmethod
+    def simulate_get_request(cls, locale, domain, locale_dir):
+        with override(locale):
+            catalog_view = cls()
+            catalog_view.locale_paths = [locale_dir]
+            response = catalog_view.get(
+                request=None,  # we are passing None as the request, as the request
+                               # object is currently not used by django
+                packages='dummy_package',  # This variable is only passed so that
+                                           # the `get_paths` method can be overridden.
+                domain=domain,
+            )
+            return force_str(response.content)
+
+
+def create_js_namespaced_catalog(xblock_conf_locale_dir, locale, source_mo_file, namespace, output):
+    """
+    Compile .mo to .js gettext catalog similar to the `compilejsi18n` command.
```

Comment:
Check the license and consider contributing this function into https://github.com/zyegfryed/django-statici18n instead of keeping it here.